lib/transactional.pm

```diff
@@ -176,3 +177,28 @@ sub ensure_rollback_service_not_running {
         $output =~ '^(start|running)$' ? sleep 10 : last;
     }
 }
+
+
+# Verify that essential snapshots cannot be deleted, see https://jira.suse.com/browse/SLE-3804
+sub check_undel_snaps {
+    # Create some snapshots with a transactional update snapshot in the middle.
+    record_info "Check undel", "Check that essential snapshots cannot be deleted, creates snapshots and installs a package";
+    assert_script_run("snapper create");
+    trup_call "ptf install" . rpmver('security');
+    assert_script_run("snapper create");
+
+    my $last_snapshot        = +(split /\n/, script_output('snapper list --disable-used-space'))[-1];
+    my $last_snapshot_number = +(split ' ',  $last_snapshot)[0];
+    my @expected        = ('current system', 'currently mounted', 'to be mounted');
+    my @delete_response = (split /\n/,       script_output("snapper delete 0-$last_snapshot_number 2>&1"));
+
+    for my $line (0 .. $#expected) {
+        record_info("$expected[$line]", "Snapper delete response: $delete_response[$line]");
+        unless ($delete_response[$line] =~ /^Cannot\sdelete.*$expected[$line].*/) {
+            die("Deletion of essential snapshot not prevented ? Expected: \"Cannot delete...$expected[$line]...\" got: \"$delete_response[$line]\"");
+        }
+        $line++;
+    }
+    my @snapshots_after = (split /\n/, script_output('snapper list --disable-used-space'));
+    die("Incorrect number of snapshots after deletion") unless (@snapshots_after == 5);
```

Comment:
As always, avoiding magic numbers helps readability. Could you assign it to a variable?

Reply:
this =5 is hacky because you are counting the header, I guess we need some regex or more simple way to count them.

Reply:
Guys, I think you're just being picky. I'm not gonna change this, it works, and no it's not a "magic" number.

Reply:
https://en.wikipedia.org/wiki/Magic_number_(programming)

Reply:
> https://en.wikipedia.org/wiki/Magic_number_(programming)

Still not a magic number: after deleteing all the snapshots you should end with exactly 5 lines, it's not a random value that may change. I can put a comment if you prefer.

Reply:
I mean, I can declare my numberoflines=5 and then use this, but really it's pointless here. I think the code is cleare enough 
die("**Incorrect number of snapshots after deletion"**) unless (@snapshots_after == 5);